dev-client/src/model/soilId/actions/soilDataActions.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {User} from 'terraso-client-shared/account/accountSlice';
+import {
+  DepthDependentSoilDataUpdateMutationInput,
+  SoilDataDeleteDepthIntervalMutationInput,
+  SoilDataUpdateDepthIntervalMutationInput,
+  SoilDataUpdateMutationInput,
+} from 'terraso-client-shared/graphqlSchema/graphql';
+import * as remoteSoilData from 'terraso-client-shared/soilId/soilDataService';
+import {SoilData} from 'terraso-client-shared/soilId/soilIdTypes';
+import {ThunkAPI} from 'terraso-client-shared/store/utils';
+
+import {isFlagEnabled} from 'terraso-mobile-client/config/featureFlags';
+import * as localSoilData from 'terraso-mobile-client/model/soilId/actions/localSoilDataActions';
+import {AppState} from 'terraso-mobile-client/store';
+
+export const updateSoilDataThunk = async (
```

Comment:
Unsure if it makes more sense to just merge the thunk methods with the non-thunk methods -- I suspect that the separation isn't necessary, but somehow this felt cleaner to me.